lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -3219,6 +3219,118 @@ void mme_app_handle_nw_init_bearer_deactv_req(
   OAILOG_FUNC_OUT(LOG_MME_APP);
 }
 
+void mme_app_handle_handover_required(
+    mme_app_desc_t* mme_app_desc_p,
+    itti_s1ap_handover_required_t* const handover_required_p) {
+  OAILOG_FUNC_IN(LOG_MME_APP);
+
+  struct ue_mm_context_s* ue_context_p          = NULL;
+  MessageDef* message_p                         = NULL;
+  itti_mme_app_handover_request_t* ho_request_p = NULL;
+
+  OAILOG_INFO(
+      LOG_MME_APP,
+      "Received HANDOVER_REQUIRED from S1AP for "
+      "ue-id " MME_UE_S1AP_ID_FMT "\n",
+      handover_required_p->mme_ue_s1ap_id);
+
+  ue_context_p =
+      mme_ue_context_exists_mme_ue_s1ap_id(handover_required_p->mme_ue_s1ap_id);
+
+  if (!ue_context_p) {
+    OAILOG_ERROR(
+        LOG_MME_APP, "UE context doesn't exist for UE " MME_UE_S1AP_ID_FMT "\n",
+        handover_required_p->mme_ue_s1ap_id);
+  }
+
+  message_p    = itti_alloc_new_message(TASK_MME_APP, MME_APP_HANDOVER_REQUEST);
+  ho_request_p = &message_p->ittiMsg.mme_app_handover_request;
+
+  // get the ue security capabilities
+  ho_request_p->encryption_algorithm_capabilities =
+      ((uint16_t) ue_context_p->emm_context._ue_network_capability.eea &
+       ~(1 << 7))
+      << 1;
+  ho_request_p->integrity_algorithm_capabilities =
+      ((uint16_t) ue_context_p->emm_context._ue_network_capability.eia &
+       ~(1 << 7))
+      << 1;
+
+  // copy information from the handover required message
+  ho_request_p->mme_ue_s1ap_id       = handover_required_p->mme_ue_s1ap_id;
+  ho_request_p->target_sctp_assoc_id = handover_required_p->sctp_assoc_id;
+  ho_request_p->target_enb_id        = handover_required_p->enb_id;
+  ho_request_p->cause                = handover_required_p->cause;
+  ho_request_p->handover_type        = handover_required_p->handover_type;
+  ho_request_p->src_tgt_container    = handover_required_p->src_tgt_container;
```

Comment:
You are passing the ownership of `src_tgt_container` once more. This kind of breaks the contract of message passing (i.e., sender allocates, receiver frees). Let's do a bstring copy here and also make additions to `itti_free_msg_content` to free bstrings and other pointer types for all the new message types defined by this feature.